extern/boostd-data/yugabyte/service.go

```diff
@@ -442,54 +450,63 @@ func (s *Store) addMultihashesToPieces(ctx context.Context, pieceCid cid.Cid, re
 	ctx, span := tracing.Tracer.Start(ctx, "store.add_index.payloadpiece")
 	defer span.End()
 
-	var count float64
-	return s.execParallel(ctx, recs, s.settings.PayloadPiecesParallelism, func(rec model.Record) error {
-		multihashBytes := rec.Cid.Hash()
-		q := `INSERT INTO PayloadToPieces (PayloadMultihash, PieceCid) VALUES (?, ?)`
-		err := s.session.Query(q, trimMultihash(multihashBytes), pieceCid.Bytes()).Exec()
-		if err != nil {
-			return fmt.Errorf("inserting into PayloadToPieces: %w", err)
+	insertPieceOffsetsQry := `INSERT INTO PayloadToPieces (PayloadMultihash, PieceCid) VALUES (?, ?)`
+	pieceCidBytes := pieceCid.Bytes()
+
+	var batch *gocql.Batch
+	for allIdx, rec := range recs {
+		if batch == nil {
+			batch = s.session.NewBatch(gocql.UnloggedBatch).WithContext(ctx)
+			batch.Entries = make([]gocql.BatchEntry, 0, s.settings.InsertBatchSize)
```

Comment:
Size should be whatever is left, which is _at most_ `insertBatchSize`, not always.